opencompass/summarizers/multi_model.py

```diff
@@ -0,0 +1,374 @@
+                    raw_results[model_abbr][sg['name']] = {'error': 'missing datasets: {}'.format(set(sg['subsets']) - set(results.keys()))}
+
+        prompt_version = {dataset_abbr_from_cfg(d): get_prompt_hash(d)[:6] for d in dataset_cfgs}
+
+        # format table
+        summarizer_dataset_abbrs = []
+        if self.dataset_abbrs is None:
+            for dataset in dataset_cfgs:
+                dataset_abbr = dataset_abbr_from_cfg(dataset)
+                if dataset_abbr in dataset_metrics:
+                    for metric in dataset_metrics[dataset_abbr]:
+                        summarizer_dataset_abbrs.append((dataset_abbr, metric))
+                else:
+                    summarizer_dataset_abbrs.append((dataset_abbr, None))
+            for dataset_abbr in dataset_metrics:
+                for metric in dataset_metrics[dataset_abbr]:
+                    if (dataset_abbr, metric) not in summarizer_dataset_abbrs:
+                        summarizer_dataset_abbrs.append((dataset_abbr, metric))
+        else:
+            for item in self.dataset_abbrs:
+                if isinstance(item, str):
+                    summarizer_dataset_abbrs.append((item, None))
+                elif isinstance(item, (list, tuple)):
+                    summarizer_dataset_abbrs.append((item[0], item[1]))
+
+        table = []
+        header = ['dataset', 'version', 'metric', 'mode'] + model_abbrs
+        table.append(header)
+        for dataset_abbr, metric in summarizer_dataset_abbrs:
+            if dataset_abbr not in dataset_metrics:
+                table.append([dataset_abbr, '-', '-', '-'] + ['-'] * len(model_abbrs))
+                continue
+            if metric is None:
+                index = 0
+                metric = dataset_metrics[dataset_abbr][0]
+            elif metric in dataset_metrics[dataset_abbr]:
+                index = dataset_metrics[dataset_abbr].index(metric)
+            else:
+                table.append([dataset_abbr, '-', '-', '-'] + ['-'] * len(model_abbrs))
+                continue
+
+            row = [dataset_abbr, prompt_version.get(dataset_abbr, '-'), metric, dataset_eval_mode.get(dataset_abbr, '-')]
+            for model_abbr in model_abbrs:
+                if dataset_abbr in parsed_results[model_abbr]:
+                    row.append('{:.02f}'.format(parsed_results[model_abbr][dataset_abbr][index]))
+                else:
+                    row.append('-')
+            table.append(row)
+
+        self.models_summary_group_metrics[table[0][-1]] = summary_group_metrics
+        return table
+
+    def merge(self, summarizer: 'MultiModelSummarizer'):
+        assert len(self.table) == len(summarizer.table)
+        for row_i, row in enumerate(summarizer.table):
+            base_row = self.table[row_i]
+            if base_row[:3] != row[:3]:
+                self.logger.warning(f'cannot merge tables with different headers: {base_row} vs {row}')
+            base_row.append(row[-1])
```

Comment:
One config file can have multiple models to summarize
```suggestion
            base_row.extend(row[4:])
```